lib/actions/user.js

```diff
@@ -55,19 +63,21 @@ export function fetchOrInitializeUser (auth) {
       // }
       // TODO: Improve AWS response.
 
-      const resultData = result.data
-      const isNewAccount = result.status === 'error' || (resultData && resultData.result === 'ERR')
-
+      const isNewAccount = fetchUserStatus === 'error' || (user && user.result === 'ERR')
       if (!isNewAccount) {
         // TODO: Move next line somewhere else.
-        if (resultData.savedLocations === null) resultData.savedLocations = []
-        dispatch(setCurrentUser({ accessToken, user: resultData }))
+        if (user.savedLocations === null) user.savedLocations = []
+
+        // Load user's monitored trips before setting the user state.
+        const { data: trips, status: tripFetchStatus } = await getTrips(otpMiddleware, accessToken)
+        if (tripFetchStatus === 'success') {
+          dispatch(setCurrentUserMonitoredTrips(trips))
+        }
+
+        dispatch(setCurrentUser({ accessToken, user }))
       } else {
-        dispatch(setCurrentUser({ accessToken, user: getStateForNewUser(user) }))
+        dispatch(setCurrentUser({ accessToken, user: getStateForNewUser(authUser) }))
```

Comment:
`getStateForNewUser` should probably be called `constructNewUser` or something, right?

Reply:
Addressed.